crates/ruff_python_formatter/src/builders.rs

```diff
@@ -323,6 +324,57 @@ impl<'fmt, 'ast, 'buf> JoinCommaSeparatedBuilder<'fmt, 'ast, 'buf> {
     }
 }
 
+/// Format comments inside empty parentheses, brackets or curly braces.
+///
+/// Empty `()`, `[]` and `{}` are special because there can be dangling comments, and they can be in
+/// two positions:
+/// ```python
+/// x = [  # end-of-line
+///     # own line
+/// ]
+/// ```
+/// These comments are dangling because they can't be assigned to any element inside as they would
+/// in all other cases.
+pub(crate) fn empty_with_dangling_comments(
```

Comment:
Nit: Maybe `empty_parenthesized_with_dangling_comments`

Reply:
added this, we need to figure out a better nomenclature for `(` vs. {`(`, `[`, `{`}